gnd/src/main/java/com/google/android/gnd/ui/editobservation/EditObservationViewModel.java

```diff
@@ -193,6 +212,52 @@ public void onFocusChange(Field field, boolean hasFocus) {
     }
   }
 
+  void initPhotoSelector(String fieldId) {
+    /*
+     * Didn't subscribe this with Fragment's lifecycle because we need to retain the disposable
+     * after the fragment is destroyed (for activity result)
+     */
+    disposeOnClear(
+        storageManager
+            .launchImagePicker()
+            .andThen(
+                storageManager
+                    .imagePickerResult()
```

Comment:
Can this and the other lambdas blocks below be refactored into methods with clear names to make it easier to follow?